tests/cephfs/cephfs_mirroring/cephfs_mirroring_utils.py

```diff
@@ -742,6 +744,87 @@ def list_and_verify_remote_snapshots_and_data(
+        """
+        try:
+            mounting_dir = "".join(
+                random.choice(string.ascii_lowercase + string.digits)
+                for _ in list(range(10))
+            )
+            target_mount_path = f"/mnt/{mounting_dir}"
+            target_clients.exec_command(sudo=True, cmd=f"mkdir -p {target_mount_path}")
+            target_clients.exec_command(
+                sudo=True,
+                cmd=f"ceph-fuse -n {target_client_user} {target_mount_path} --client_fs {target_fs_name}",
+            )
+            snapshot_list_command = f"ls {target_mount_path}{source_path}.snap"
+            retry_cmd = retry(CommandFailed, tries=3, delay=30)(
+                target_clients.exec_command
+            )
+            snapshots, _ = retry_cmd(sudo=True, cmd=snapshot_list_command)
+            snapshots = snapshots.strip().split()
+            log.info(f"Available Snapshots : {snapshots}")
+
+            out_target, rc = target_clients.exec_command(
+                sudo=True,
+                cmd=f"bash /root/md5sum_script.sh {target_mount_path}{source_path}",
+            )
+            log.info(f"Checksums of the files in target cluster : \n {out_target}")
+            out_source, rc = source_client.exec_command(
+                sudo=True,
+                cmd=f"bash /root/md5sum_script.sh {source_mount_path}{source_path}",
+            )
+            log.info(f"Checksums of the files in source cluster : \n {out_source}")
+            if out_target == out_source:
+                log.info("Checksums are matching and all files synced")
+            out_source_snap, rc = target_clients.exec_command(
+                sudo=True,
+                cmd=f"bash /root/md5sum_script.sh {target_mount_path}{source_path}/.snap",
+            )
+            log.info(
+                f"Checksums of the files in source cluster snap dir : \n {out_source_snap}"
+            )
+            out_target_snap, rc = source_client.exec_command(
+                sudo=True,
+                cmd=f"bash /root/md5sum_script.sh {source_mount_path}{source_path}/.snap",
+            )
+            log.info(
+                f"Checksums of the files in target cluster snap dir : \n {out_target_snap}"
+            )
+            if out_target_snap != out_source_snap:
+                return False, "Checksums are not matching in snapshot folders"
+
+            log.info("Checksums are matching and all files synced in snapshot folder")
+
+            if out_target != out_source:
+                return False, "Checksums are not matching in subvolume folders"
+            log.info("Checksums are matching and all files synced in Subvolume folder")
+            return True, "All Files and checksums synced Properly"
+        except Exception as e:
+            return False, f"Error: {str(e)}"
+        finally:
+            target_clients.exec_command(sudo=True, cmd=f"umount {target_mount_path}")
```

Comment:
let cleanup be part of another function.
Sometimes we might need to run a few scenarios after verifying on remote.

Reply:
This mount we are doing as part of this function only so I have added cleanup here.
We are not touching cluster data(data synced). We are unmounting and deleting the mounted path created